TTlibrary/config/routes.rb

```diff
@@ -2,13 +2,16 @@
   devise_for :users
   root 'books#home'
 
+  resources :users do
+    resources :requests
+  end
+  
   resources :authors do
     resources :books do
-      resources :comments
+      resources :comments, :requests
```

Comment:
It isn't necessary to declare :requests here. Remember to add the book_id param to the "Request a book" link (and change the route of the link).